geps/gep-2162.md

```diff
@@ -0,0 +1,180 @@
+
+Note: implementations must keep the published list sorted.
+
+## API
+
+This GEP proposes API changes describes as follow:
+
+* Update the `GatewayClassStatus` struct to include a string-represented list of `SupportedFeatures`.
+
+
+```go
+// GatewayClassStatus is the current status for the GatewayClass.
+type GatewayClassStatus struct {
+    // Conditions is the current status from the controller for
+    // this GatewayClass.
+    //
+    // Controllers should prefer to publish conditions using values
+    // of GatewayClassConditionType for the type of each Condition.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // SupportedFeatures is the features the GatewayClass support.
+    // <gateway:experimental>
+    // +kubebuilder:validation:MaxItems=64
+    SupportedFeatures []string `json:"supportedFeatures,omitempty"`
+}
+```
+
+## Understanding SupportedFeatures field
+
+Its important to define how we read the list of `SupportedFeatures` we report.
+
+We have no support features for core features. If an implementation reports a route name e.g `HTTPRoute` as a supportedFeature it means it supports all its core features.
+In other words, supporting the route's core features is a requirement for the implementation to say that it supports the route.
+
+For Extended/Implementation-specific features we have the supported features names.
+
+An example of a GatewayClass Status with the SupportedFeatures reported would look like:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+...
+status:
+  conditions:
+  - lastTransitionTime: "2022-11-16T10:33:06Z"
+    message: Handled by XXX controller
+    observedGeneration: 1
+    reason: Accepted
+    status: "True"
+    type: Accepted
+  supportedFeatures:
+    - HTTPRoute
+    - HTTPRouteHostRewrite
+    - HTTPRoutePortRedirect
+    - HTTPRouteQueryParamMatching
+
+```
+## Standardize features and conformance tests names
+
+Before we add the supported features into our API, it is necessary to establish standardized naming and formatting conventions.
+
+### Formatting Proposal
+
+#### Feature Names
+
+Every feature should:
+
+1. Start with the resource name. i.e HTTPRouteXXX
+2. Follow the PascalCase convention. Note that the resource name in the string should come as is and not be converted to PascalCase, i.e HTTPRoutePortRedirect and not HttpRoutePortRedirect.
+3. Not exceed 128 characters.
+4. Contain only letters and numbers
+
+#### Conformance tests names
+
+Conformance tests file names should try to follow the the `pascal-case-name.go` format.
+For example for `HTTPRoutePortRedirect` - the test file would be `httproute-port-redirect.go`.
+
+We should treat this requirement as "best effort" because we might have test files that checks combination of features and can't follow the same format.
```

Comment:
```suggestion
We should treat this guidance as "best effort" because we might have test files that check the combination of several features and can't follow the same format.
```